File: BlockChainWeb/Helpers/BlockHelpers.cs
using BlockChain;
using static BlockChain.Types;

namespace BlockChainWeb.Helpers
{
    public class BlockHelpers
    {
        public static Types.BlockWithHash DtoToBlock(Repository.Block lastBlockDto)
        {
            var block = new Types.Block(lastBlockDto.index, lastBlockDto.minedBy, lastBlockDto.data, lastBlockDto.previousHash, lastBlockDto.nonce);
            return new Types.BlockWithHash(block, lastBlockDto.hash);
        }

        public static Repository.Block BlockToDto(BlockWithHash blockWithHash)
        {
            return new Repository.Block()
            {
                index = blockWithHash.block.index,
                data = blockWithHash.block.data,
                minedBy = blockWithHash.block.minedBy,
                nonce = blockWithHash.block.nonce,
                previousHash = blockWithHash.block.previousHash,
                hash = blockWithHash.hash
            };
        }
    }
}

File: Code/BlockChainMiner/BlockHelpers.fs
module BlockChain.BlockHelpers
open BlockChain.Types
open Repository

let DtoToBlock (lastBlockDto:Repository.BlockDto ) =
    {
        block = {
                index = lastBlockDto.index;
                minedBy = lastBlockDto.minedBy;
                data = lastBlockDto.data;
                previousHash = lastBlockDto.previousHash
                nonce = lastBlockDto.nonce
            }
        hash = lastBlockDto.hash
    }

let BlockToDto(blockWithHash:BlockWithHash) =
    new Repository.BlockDto(
        blockWithHash.block.index,
        blockWithHash.block.minedBy,
        blockWithHash.block.data,
        blockWithHash.block.nonce,
        blockWithHash.block.previousHash,
        blockWithHash.hash);
